tasks/main.yml

```diff
@@ -18,49 +18,85 @@
 
 - name: Install and license Cobalt Strike
   block:
-    - name: Grab Cobalt Strike tarball and license from S3
+    - name: Grab Cobalt Strike license from S3
       amazon.aws.aws_s3:
         bucket: "{{ bucket_name }}"
-        object: "{{ item }}"
-        dest: "/tmp/{{ item }}"
+        object: "{{ license_object_name }}"
+        dest: "/tmp/cobaltstrike.license"
         mode: get
       become: no
       delegate_to: localhost
-      loop:
-        - "{{ tarball_object_name }}"
-        - "{{ license_object_name }}"
 
-    - name: Copy the Cobalt Strike tarball
-      ansible.builtin.copy:
-        src: /tmp/{{ tarball_object_name }}
-        dest: /tmp/{{ tarball_object_name }}
-        mode: 0644
+    - name: Install curl
+      ansible.builtin.package:
+        name:
+          - curl
+
+    # I'd prefer to use ansible.builtin.uri here, but the following
+    # code doesn't work with the Cobalt Strike site:
+    # - name: Get download token from Cobalt Strike website
+    #   ansible.builtin.uri:
+    #     body: dlkey={{ cobaltstrike_license }}
+    #     body_format: "form-urlencoded"
+    #     dest: /tmp/html.html
+    #     method: POST
+    #     return_content: yes
+    #     url: https://www.cobaltstrike.com/download
+    #     # grep --fixed-strings 'href="/downloads/' |
+    #     # cut --delimiter='/' --fields=3
```

Comment:
Is this just here to reference what you'd need to do next?

Reply:
Please see commit c1900e4.